### PR TITLE
Fix #12655 copy/paste style between different vector layer geometry types

### PR DESCRIPTION
**Pull request by Hvitnov**
User now gets qMessagebox message when trying to copy/paste styles between layers with different geometry types. Earlier behavior caused the destination layer of the paste to dissappear from the viewport

### CODE REVIEW

**Comment by pcav** [comment]
Looks an important fix. Could someone revise it for inclusion before feature freeze (if this can be considered a feature)?
@wonder-sk

File: src/app/qgisapp.cpp [hunk 1]
```diff
@@ -6515,12 +6536,19 @@ void QgisApp::copyStyle( QgsMapLayer * sourceLayer )
     mActionPasteStyle->setEnabled( true );
   }
 }
+/**
+   \param destinatioLayer  The layer that the clipboard will be pasted to
```
Comment (nyalldawson) [inline src/app/qgisapp.cpp]: should be "destinationLayer"

**Comment by alexbruy** [comment]
Maybe it is better to use message bar instead of message box?

**Comment by NathanW2** [comment]
Yes please. No message boxes.

**Comment by NathanW2** [comment]
A good solution in the future might be to auto convert over the styles even if you loose some of the information in the process.

**Comment by Hvitnov** [comment]
ErrorBox was used in copyStyle and pasteStyle already, but is now changed to errorbar.

Thanks for the feedback.
A future solution could be interesting. Intuitively I personally would expect a behavior where only the part of the style relevant for the layer pasted to was actually pasted. For instance pasting a line style to a polygon could change only the style of the edges/lines in the polygon layer.

File: src/app/qgisapp.cpp (hunk 1 above)
Comment (Hvitnov) [inline src/app/qgisapp.cpp]: This is also fixed in the latest version

**Pull request merged**